token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenService.java

```diff
@@ -73,10 +71,10 @@ protected OAuth2TokenResponse requestAccessToken(URI tokenEndpointUri, HttpHeade
 
 	private OAuth2TokenResponse executeRequest(HttpPost httpPost) throws OAuth2ServiceException {
 		httpPost.addHeader(USER_AGENT, HttpClientUtil.getUserAgent());
-		LOGGER.debug("Requesting access token from url {} with headers {}", httpPost.getURI(),
-				httpPost.getAllHeaders());
+		LOGGER.debug("Requesting access token from url {} with headers {}", httpPost.getRequestUri(),
+				httpPost.headerIterator());
 		try (CloseableHttpResponse response = httpClient.execute(httpPost)) {
```

Comment:
Since Apache HTTP Client 5, one should use non-deprecated execute method by supplying a Response Handler. As discussed, we will overhaul these calls in the next commits.